frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -259,18 +296,58 @@ export default ({
               // when we don't get event after a certain period
               pending: true
             })
+            this.stopReplying()
+            this.updateScroll()
           }
         }
       })
-      // need to scroll to the bottom
-      this.updateScroll()
     },
-    updateScroll () {
+    async scrollToMessage (messageId, focus = true) {
+      if (!messageId) {
+        return
+      }
+
+      const scrollAndHighlight = (index) => {
+        const eleMessage = document.querySelectorAll('.c-body-conversation > .c-message')[index]
```

Comment:
Could you use ref here to access the Dom here?

Reply:
Sorry @pieer, I have tried already.
I couldn't get the expected value of `allElements[i].offsetParent.offsetTop`.

```
const allElements = document.querySelectorAll('.c-body-conversation > .c-message')
for (let i = allElements.length - 1; i >= 0; i--) {
  if (allElements[i].offsetTop - allElements[i].offsetParent.offsetTop < curScrollTop) {
    sbp('state/vuex/commit', 'setChatRoomScrollPosition', {
      chatRoomId: this.currentChatRoomId,
      messageId: this.messages[i].id
    })
    break
  }
}
```

Can you please give me some idea about it?

Reply:
The ref should be on the component
component(is='messageType' ref='message.id' offset=''...)
Then loop on messages to get already calculted offset for quick comparison between curScrollTop and this.refs[message.id].offset

Reply:
```
component(
          :is='messageType(message)'
          :ref='message.id'
          ...
)
```
After define `ref` like the above, I can access `offsetTop` of each element in the following way.

`this.$refs[m.id][0].$el.offsetTop`

Is there any shortcut to the offsetTop? `this.$refs[m.id][0].$el.offsetTop` seems too complicated.